india_compliance/gst_india/setup/__init__.py

```diff
@@ -195,3 +199,47 @@ def _get_custom_fields_to_create(*custom_fields_list):
             result.setdefault(doctypes, []).extend(fields)
 
     return result
+
+
+def mapp_default_uoms():
+    # Mapping of GST UOMs with ERPNext UOMs
+    settings = frappe.get_doc("GST Settings")
+
+    if not frappe.db.table_exists("GST UOM Mapping"):
+        return
+
+    for _gst_uom in GST_UOMS:
+        # get uom from GST UOM String to match in UOM table
+        gst_uom = get_uoms_to_match(_gst_uom)[:-1]
+
+        # validate if gst uom exists in UOMs
+        default_uoms = frappe.db.get_all(
+            "UOM", filters={"name": ["like", f"{gst_uom}"]}, fields=["name"]
+        )
+
+        if not default_uoms:
+            continue
+
+        for uom in default_uoms:
+            # validate if uom is already mapped
+            if validate_if_uom_is_mapped(settings, uom.name):
+                continue
+            settings.append("gst_uom_mapping", get_mapped_uom(_gst_uom, uom.name))
+
+    settings.save()
+
+
+def validate_if_uom_is_mapped(settings, uom):
+    return next(
+        (True for mapping in settings.gst_uom_mapping if mapping.uom == uom), False
+    )
+
+
+def get_uoms_to_match(uom):
+    return UOM_REGEX.search(uom).group(1)
```

Comment:
May not be appropriate.

First, check for an exact match if available.
Hardcode other matches and use them to update.